core/src/main/java/nl/nn/adapterframework/pipes/UnzipPipe.java

```diff
@@ -216,25 +215,26 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 								throw new PipeRunException(this, "file [" + tmpFile.getAbsolutePath() + "] already exists"); 
 							}
 						} else {
-							String filename = new File(basename).getName();
-							String zipEntryPath = entryname.substring(0, ze.getName().indexOf(filename));
-							if(filename.length() < 3) filename += ".tmp.";	//filename here is a prefix to create a unique filename and that prefix must be at least 3 chars long
-							tmpFile = File.createTempFile(filename, extension, new File(dir, zipEntryPath));
+							if (isCreateSubdirectories()) {
```

Comment:
Is createSubDirectories only used when keepOriginalFilename=false?
Could you create tests for all combinations of options?